### PR TITLE
Improve robustness of TCP connector

### PR DESCRIPTION
**Pull request by mavam**
This PR streamlines the response promise handling in the TCP connector, avoiding potential double delivery.

### CODE REVIEW

File: libtenzir/builtins/connectors/tcp.cpp [hunk 1]
```diff
@@ -131,24 +130,26 @@ auto make_tcp_bridge(tcp_bridge_actor::stateful_pointer<tcp_bridge_state> self)
...
                     ec::system_error,
                     fmt::format("connection failed: {}", ec.message())));
+                }
                 if (self->state.tls_socket) {
                   self->state.tls_socket->handshake(
                     boost::asio::ssl::stream<
                       boost::asio::ip::tcp::socket>::client,
                     ec);
+                  if (ec) {
+                    return self->state.connection_rp.deliver(caf::make_error(
```
Comment (tobim) [inline libtenzir/builtins/connectors/tcp.cpp:144]: IIuc this `return` is the only non-cosmetic change, is that correct?

**Review by tobim — COMMENTED** [review]
Can you provide an example pipeline that now behaves differently? Ideally in the form of a test ;).

File: libtenzir/builtins/connectors/tcp.cpp (hunk 1 above)
Comment (mavam) [reply libtenzir/builtins/connectors/tcp.cpp:144]: Some `return`s are cosmetic transformations for consistency throughout a single function, others not. For example, a failed TLS handshake caused a double delivery.

**Comment by mavam** [comment]
> Can you provide an example pipeline that now behaves differently? Ideally in the form of a test ;).

Alas, no. It's a bit stabbing in the dark. At least one potential bug is fixed with this PR, the rest are transformations that result in a more robust control flow.

**Review by tobim — APPROVED** [review]

**Pull request merged**